core/orchestrator.go

```diff
@@ -199,6 +196,15 @@ func (orch *orchestrator) PriceInfo(sender ethcommon.Address) (*big.Rat, error)
 	return new(big.Rat).Mul(orch.node.PriceInfo, overhead), nil
 }
 
+// SufficientBalance checks whether the credit balance for a stream is sufficient
+// to proceed with downloading and transcoding
+func (orch *orchestrator) SufficientBalance(manifestID ManifestID) bool {
```

Comment:
Just realized I missed this in the original code review - currently, this method will break off-chain mode because `orch.node.Balances` will be be nil in off-chain mode. We should add a guard clause in this method that returns `true` if `orch.node`, `orch.node.Balances` or `orch.node.Recipient` are nil similar to the guard clause in [ProcessPayment()](https://github.com/livepeer/go-livepeer/blob/master/core/orchestrator.go#L99).

Reply:
You're right, I noticed this when firing up nodes, but forgot this over the weekend.

Reply:
Should be dealt with in https://github.com/livepeer/go-livepeer/pull/956/commits/59957ab90df4921cb86674348b1c78374098c0b0